folium/features.py

```diff
@@ -1095,6 +1095,9 @@ def get_by_key(obj, key):
 
             def color_scale_fun(x):
                 key_of_x = get_by_key(x, key_on)
+                if not key_of_x:
+                    if len(key_on) > 0:
+                        raise("key_on value not found in geojson")
```

Comment:
Would be really helpful if we could include some information in the error message on what's wrong with the data. You could for example show the `key_on` value in the error message. Something like `'wrong value: {}'.format(value)`.